test/views/UnderpaymentTypeViewSpec.scala

```diff
@@ -0,0 +1,118 @@
+/*
```

Comment:
Review other view specs and refactor accordingly. Right now you're missing the tests that assert no errors are displayed when no errors exist. Also, other view specs separate fixed content checks into a separate group of tests (look for `it should {...`)